extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java

```diff
@@ -67,6 +68,7 @@ public void handle(final RoutingContext ctx) {
         }
         try {
             handleWithIdentity(ctx);
+            currentManagedContext.deactivate();
```

Comment:
Why isn't this one in a finally?  Why deactivate in one case and terminate in the case of an exception?

Also what's the purpose of this.currentManagedContextTerminationHandler now?

Reply:
`this.currentManagedContextTerminationHandler ` will terminate (so deactivate and destroy) once we are done, while the deactivate after the `handleWithIdentity` will only deactivate (so that it does not leak)

Reply:
It does not need to be in a finally as the catch with terminate

Reply:
I can understand that but my concern was more that we are not consistent. Typically here: https://github.com/quarkusio/quarkus/pull/27040/files#diff-5b5cda47e6a5418c9345a577d468f058e05c412dc9b64fce6a6ef5aec496023bR75-R85 where we simply deactivate. Or maybe the exception from the latter is caught below?

Also what do you mean by leaking? What I don't understand is how we can leak something between this and the end of the response? Or is it that the thread is somehow reused between the end of this and the end of the response?

Reply:
The thread could be reused, in a high traffic scenario.

Reply:
But you're sure that all the response handlers we added earlier are always executed on the right thread at the right time?
That's what gets me a bit worried.

Note that I have no idea how this all works so I'm asking naive questions.

Reply:
Yes sure. Maybe @stuartwdouglas can explain better. By leaking I mean that a new request that use the same thread can get the context that still float around (because it's not been deactivated)